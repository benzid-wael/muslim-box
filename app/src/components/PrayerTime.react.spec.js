import React from "react";
import { mount, shallow } from "enzyme";
import assert from "assert";
import timezoneMock from "timezone-mock";

import PrayerTime from "./PrayerTime.react";

describe("<PrayerTime />", () => {
  const startTimestamp = 1658326500; // 2022-07-20T14:17:00.000Z
  const secondsInOneHour = 3600;
  const endTimestamp = startTimestamp + secondsInOneHour;
  const prayer: PrayerTime = {
    name: "Fajr",
    isPrayer: true,
    start: startTimestamp,
    end: endTimestamp,
    visible: true,
  };

  beforeAll(() => {
    timezoneMock.register("UTC");
  });

  afterAll(() => {
    timezoneMock.unregister();
  });

  beforeEach(() => {
    // We are setting the current time to 17m before end time
    Date.now = jest.fn(() => new Date("2022-07-20T15:00:00.000Z"));
  });

  it("<PrayerTime /> when prayer is not active", () => {
    const container = shallow(<PrayerTime prayer={prayer} />);
    expect(container.html()).toMatchSnapshot();
  });

  it("<PrayerTime /> when prayer is active", () => {
    const container = shallow(<PrayerTime prayer={prayer} isCurrent={true} />);
    expect(container.html()).toMatchSnapshot();
  });

  it("<PrayerTime /> when reminder is active", () => {
    // to activate the reminder, we need to set endTimeReminderInMinutes
    // to more than 17m as it remains only 17m till the prayer end time
    // (check start, end and mocked current time)
    const container = shallow(<PrayerTime prayer={prayer} isCurrent={true} endTimeReminderInMinutes={20} />);
    expect(container.html()).toMatchSnapshot();
  });
});
