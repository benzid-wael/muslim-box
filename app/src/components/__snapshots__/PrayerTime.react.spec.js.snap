// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrayerTime /> <PrayerTime /> when prayer is active 1`] = `"<div class=\\"sc-bczRLJ egLEnW\\"><div class=\\"sc-gsnTZi KeiYk\\"><div class=\\"sc-hKMtZM sc-eCYdqJ ijvmlV gbLpsd\\">Fajr</div><div class=\\"sc-hKMtZM sc-jSMfEi ijvmlV dtjjYe\\">15:15</div><div class=\\"sc-hKMtZM sc-jSMfEi sc-gKXOVf ijvmlV dtjjYe Qrzkn\\">16:15</div></div></div>"`;

exports[`<PrayerTime /> <PrayerTime /> when prayer is not active 1`] = `"<div class=\\"sc-bczRLJ eWFJjW\\"><div class=\\"sc-gsnTZi KeiYk\\"><div class=\\"sc-hKMtZM sc-eCYdqJ ijvmlV gbLpsd\\">Fajr</div><div class=\\"sc-hKMtZM sc-jSMfEi ijvmlV dtjjYe\\">15:15</div><div class=\\"sc-hKMtZM sc-jSMfEi sc-gKXOVf ijvmlV dtjjYe dFiEUT\\">16:15</div></div></div>"`;

exports[`<PrayerTime /> <PrayerTime /> when reminder is active 1`] = `"<div class=\\"sc-bczRLJ egLEnW\\"><div class=\\"sc-gsnTZi KeiYk\\"><div class=\\"sc-hKMtZM sc-eCYdqJ ijvmlV gbLpsd\\">Fajr</div><div class=\\"sc-hKMtZM sc-jSMfEi ijvmlV dtjjYe\\">15:15</div><div style=\\"color:red\\" class=\\"sc-hKMtZM sc-jSMfEi sc-gKXOVf ijvmlV dtjjYe Qrzkn\\">15:00</div></div></div>"`;
